In [1]:
import numpy as np
import pandas as pd


In [2]:
data = pd.read_csv('titanic.csv')

In [3]:
print(data.shape)

(1309, 28)


In [4]:
data.head()

,Passengerid,Age,Fare,Sex,sibsp,zero,zero.1,zero.2,zero.3,zero.4,...,zero.12,zero.13,zero.14,Pclass,zero.15,zero.16,Embarked,zero.17,zero.18,2urvived
0,1,22.0,7.2500,0,1,0,0,0,0,0,...,0,0,0,3,0,0,2.0,0,0,0
1,2,38.0,71.2833,1,1,0,0,0,0,0,...,0,0,0,1,0,0,0.0,0,0,1
2,3,26.0,7.9250,1,0,0,0,0,0,0,...,0,0,0,3,0,0,2.0,0,0,1
3,4,35.0,53.1000,1,1,0,0,0,0,0,...,0,0,0,1,0,0,2.0,0,0,1
4,5,35.0,8.0500,0,0,0,0,0,0,0,...,0,0,0,3,0,0,2.0,0,0,0


In [5]:
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 28 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Passengerid  1309 non-null   int64  
 1   Age          1309 non-null   float64
 2   Fare         1309 non-null   float64
 3   Sex          1309 non-null   int64  
 4   sibsp        1309 non-null   int64  
 5   zero         1309 non-null   int64  
 6   zero.1       1309 non-null   int64  
 7   zero.2       1309 non-null   int64  
 8   zero.3       1309 non-null   int64  
 9   zero.4       1309 non-null   int64  
 10  zero.5       1309 non-null   int64  
 11  zero.6       1309 non-null   int64  
 12  Parch        1309 non-null   int64  
 13  zero.7       1309 non-null   int64  
 14  zero.8       1309 non-null   int64  
 15  zero.9       1309 non-null   int64  
 16  zero.10      1309 non-null   int64  
 17  zero.11      1309 non-null   int64  
 18  zero.12      1309 non-null   int64  
 19  zero.13      1309

In [6]:
data[['zero.1','zero.2']].value_counts()

zero.1  zero.2
0       0         1309
Name: count, dtype: int64

## Dropping Irrelevant Columns

The dataset contains 18 extra columns named `zero.1` through `zero.18`.
All of them contain only the value `0` across all 1309 rows with no variation,
meaning they carry zero information and were likely created due to a malformed CSV export.

These columns are dropped to keep the dataset clean and relevant.

In [7]:
zero_cols = [col for col in data.columns if col.startswith('zero')]
data.drop(columns=zero_cols, inplace=True)

print("Dropped columns:", zero_cols)
print("Remaining shape:", data.shape)

Dropped columns: ['zero', 'zero.1', 'zero.2', 'zero.3', 'zero.4', 'zero.5', 'zero.6', 'zero.7', 'zero.8', 'zero.9', 'zero.10', 'zero.11', 'zero.12', 'zero.13', 'zero.14', 'zero.15', 'zero.16', 'zero.17', 'zero.18']
Remaining shape: (1309, 9)


In [8]:
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Passengerid  1309 non-null   int64  
 1   Age          1309 non-null   float64
 2   Fare         1309 non-null   float64
 3   Sex          1309 non-null   int64  
 4   sibsp        1309 non-null   int64  
 5   Parch        1309 non-null   int64  
 6   Pclass       1309 non-null   int64  
 7   Embarked     1307 non-null   float64
 8   2urvived     1309 non-null   int64  
dtypes: float64(3), int64(6)
memory usage: 92.2 KB


In [9]:
data.rename(columns={'2urvived': 'Survived'}, inplace=True)
data.head()

,Passengerid,Age,Fare,Sex,sibsp,Parch,Pclass,Embarked,Survived
0,1,22.0,7.2500,0,1,0,3,2.0,0
1,2,38.0,71.2833,1,1,0,1,0.0,1
2,3,26.0,7.9250,1,0,0,3,2.0,1
3,4,35.0,53.1000,1,1,0,1,2.0,1
4,5,35.0,8.0500,0,0,0,3,2.0,0


In [10]:
data.isnull().sum()

Passengerid    0
Age            0
Fare           0
Sex            0
sibsp          0
Parch          0
Pclass         0
Embarked       2
Survived       0
dtype: int64

In [11]:
data['Embarked'].value_counts()

Embarked
2.0    914
0.0    270
1.0    123
Name: count, dtype: int64

## Understanding the Embarked Column

The `Embarked` column indicates the port where each passenger boarded the Titanic.
Originally encoded as float values, it is mapped to actual port names for better readability:

| Code | Port |
|------|------|
| 0.0  | Cherbourg, France |
| 1.0  | Queenstown, Ireland |
| 2.0  | Southampton, England |

Southampton has the highest count (914) which matches real Titanic historical data,
as majority of passengers boarded from Southampton.

In [12]:
data['Embarked'] = data['Embarked'].map({
    0.0: 'Cherbourg',
    1.0: 'Queenstown', 
    2.0: 'Southampton'
})

data['Embarked'].value_counts()

Embarked
Southampton    914
Cherbourg      270
Queenstown     123
Name: count, dtype: int64

## Handling Missing Values in Embarked Column

Only 2 missing values found in the `Embarked` column.
Filled with mode (most frequent value) which is `Southampton`,
as it is the dominant port with 914 passengers.

In [13]:
print(data['Embarked'].isnull().sum())
data['Embarked'] = data['Embarked'].fillna(data['Embarked'].mode()[0])
print(data['Embarked'].isnull().sum())

2
0


In [14]:
data.describe()

,Passengerid,Age,Fare,Sex,sibsp,Parch,Pclass,Survived
count,1309.000000,1309.000000,1309.000000,1309.000000,1309.000000,1309.000000,1309.000000,1309.000000
mean,655.000000,29.503186,33.281086,0.355997,0.498854,0.385027,2.294882,0.261268
std,378.020061,12.905241,51.741500,0.478997,1.041658,0.865560,0.837836,0.439494
min,1.000000,0.170000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000
25%,328.000000,22.000000,7.895800,0.000000,0.000000,0.000000,2.000000,0.000000
50%,655.000000,28.000000,14.454200,0.000000,0.000000,0.000000,3.000000,0.000000
75%,982.000000,35.000000,31.275000,1.000000,1.000000,0.000000,3.000000,1.000000
max,1309.000000,80.000000,512.329200,1.000000,8.000000,9.000000,3.000000,1.000000


In [15]:
data['Age'] = data['Age'].astype(int)

In [16]:
data.rename(columns={
    'sibsp' : 'Siblings_Spouses',
    'Parch' : 'Parents_Children'
}, inplace=True)

In [17]:
data['Survived'] = data['Survived'].astype(bool)

In [18]:
data.drop(columns=['Passengerid'], inplace=True)

In [20]:
data.to_csv('titanic_cleaned.csv', index=False)
print("Saved successfully!")
data.head()

Saved successfully!


,Age,Fare,Sex,Siblings_Spouses,Parents_Children,Pclass,Embarked,Survived
0,22,7.2500,0,1,0,3,Southampton,False
1,38,71.2833,1,1,0,1,Cherbourg,True
2,26,7.9250,1,0,0,3,Southampton,True
3,35,53.1000,1,1,0,1,Southampton,True
4,35,8.0500,0,0,0,3,Southampton,False
